### ISSUE TITLE
False-positive for "Simplify check" when the check relies on overloaded true/false operators

### ISSUE THREAD

**Issue by jhinder** [dotnet/roslyn#80640]
**Version Used**: VS 18.0.0 Insiders [11104.47]

**Steps to Reproduce**:
Open the lightbulb menu on the `if` token and select "Simplify check".

```csharp
class C
{
    bool M()
    {
        if (this)
        {
            return true;
        }
        else
        {
            return false;
        }
    }

    public static bool operator true(C v) => true;

    public static bool operator false(C v) => false;
}
```

**Expected Behavior**:
The code fix is not offered as the generated code is invalid.

**Actual Behavior**:
The code is simplified to `return this`, which is invalid: `CS0029: Cannot implicitly convert type 'C' to 'bool'`

<img width="795" height="407" alt="Image" src="https://github.com/user-attachments/assets/02800022-becf-49ab-9e2e-8b81a9f87440" />

**Comment by CyrusNajmabadi** [comment]
Generally, we have not cared abou tthis case as the number of times these are overloaded in practice is massively low.  moving to backlog.
